sql: fix setting DateStyle/IntervalStyle on -c

otan opened the pull request:
Release note (bug fix): Previously, specifying IntervalStyle or
DateStyle on `options=-c...` in a pgurl would fail, even if the
sql.defaults.datestyle.enabled and sql.defaults.intervalstyle.enabled
cluster settings were set. This has now been resolved.

otan committed:
sql: fix setting DateStyle/IntervalStyle on -c

Release note (bug fix): Previously, specifying IntervalStyle or
DateStyle on `options=-c...` in a pgurl would fail, even if the
sql.defaults.datestyle.enabled and sql.defaults.intervalstyle.enabled
cluster settings were set. This has now been resolved.

rafiss reviewed (approved):
my bad i didn't see that you had this, so i wrote https://github.com/cockroachdb/cockroach/pull/72018

the TCL test is great. do you think we should go with that since it's more end-to-end, or use the TestSetSessionArguments test?

if you want you can copy my tests into this PR. i also noticed a bug with case-sensitivity. you can either add that onto here, or else feel free to merge and i'll clean up my PR later

knz reviewed (commented):
> Reviewed 2 of 2 files at r1, 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72010)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/sql/pgwire/conn_test.go, line 1469 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/72010#-Mn1eKxUAuu5fMn6-hTY:-Mn1eKxVEoSg_TvZPMAk:b6vvdt2) ([raw file](https://github.com/cockroachdb/cockroach/blob/3bd923f561228904b1d418a384b4dbdc6ef1bc48/pkg/sql/pgwire/conn_test.go#L1469)):*
> ```Go
> 			var wg sync.WaitGroup
> 			wg.Add(1)
> 			go func() {
> ```

idiomatic  (and more correct) is `defer wg.Done()` at the start of the goroutine.


<!-- Sent from Reviewable.io -->

rafiss reviewed (approved):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72010)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @otan)

___
*[pkg/sql/pgwire/conn_test.go, line 1469 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/72010#-Mn1eKxUAuu5fMn6-hTY:-Mn1jeFV5POb1ydc0oEP:bp9l96g) ([raw file](https://github.com/cockroachdb/cockroach/blob/3bd923f561228904b1d418a384b4dbdc6ef1bc48/pkg/sql/pgwire/conn_test.go#L1469)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

idiomatic  (and more correct) is `defer wg.Done()` at the start of the goroutine.
</blockquote></details>

i confirmed locally that removing the Waitgroup entirely is actually required to make this test pass

but i don't have permission to push to your fork

lgtm after you fix


<!-- Sent from Reviewable.io -->

otan committed:
sql: fix datestyle/intervalstyle passed in connection string

Release note (bug fix): Previously, session variables passed in the
connection string were case-sensitive. Now they are all correctly
normalized to lower case.
